### PR TITLE
ci: add CI, pre-commit, rustfmt, and dependabot configs

### PR DESCRIPTION
**Pull request by m4tx**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
+name: Rust CI
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: "0 0 * * *"
```
Comment (Iipin) [inline .github/workflows/rust.yml:7]: do we need to run actions in this file on a schedule as well?

**Review by Iipin — APPROVED** [review]
looks like enough goodies for bootstrapping a project, lgtm

File: .github/workflows/rust.yml (hunk 1 above)
Comment (seqre) [reply .github/workflows/rust.yml:7]: The idea behind this workflow being run on a schedule is to continuously test the codebase against updates to Rust nightly and dependencies.

File: .github/workflows/pre-commit.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Check files
+        uses: pre-commit/action@v3.0.1
```
[the remark this answers is not in the document]
Comment (seqre) [reply .github/workflows/pre-commit.yml:27]: Good point, we will implement that!

[the remark this answers is not in the document]
Comment (seqre) [reply .github/workflows/pre-commit.yml:27]: Added
image

**Pull request merged**